### ISSUE TITLE
False positive in react/prop-types

### ISSUE THREAD

**Issue by jalooc** [yannickcr/eslint-plugin-react#1623]
Eslint detects react/prop-types rule violation in the following code:

```javascript
const map = {
  one: {
    component: One,
  },
  two: {
    component: Two,
    withMargin: true,
  },
}

const Part = props => {
  if (!map[props.type]) return null

  const { component: Component, withMargin } = map[props.type]

  return <Component {...props} withMargin={withMargin} />
}

Part.propTypes = {
  type: PropTypes.string.isRequired,
}
```

```
error  '.component' is missing in props validation   react/prop-types
error  '.withMargin' is missing in props validation  react/prop-types

```

**Comment by alexzherdev** [comment]
This looks like a duplicate to #1259

**Comment by ljharb** [comment]
Yes, closing as a duplicate, thanks.
